packages/router-lite/src/route-callback.ts

```diff
@@ -0,0 +1,57 @@
+import { DI, Registration } from '@aurelia/kernel';
+import { batch } from '@aurelia/runtime';
+import { IWindow, AppTask } from '@aurelia/runtime-html';
+import { IRouterEvents, NavigationEndEvent } from './router-events';
+import { IRouter } from './router';
+import { Params, ViewportInstructionTree } from './instructions';
+
+export type IRoute<TParams = Record<string, string | undefined>, TQueryParams = Record<string, string | undefined>> = {
+    title: string;
+    path: string;
+    url: string;
+    params: TParams;
+    queryParams: TQueryParams;
+    snapshot: Partial<NavigationEndEvent>;
+  };
+  export const IRoute = DI.createInterface<IRoute>();
+
+const flattenParams = (arr: ViewportInstructionTree['children']): Readonly<Params> => {
+  const returnVal: Readonly<Params>= {} ;
+  return arr.reduce(function (done,curr){
+    return  {...done, ...curr.recognizedRoute?.route.params, ...flattenParams(curr.children)};
+    }, returnVal) ;
+};
+
+export const RouteCallback =  Registration.cachedCallback(IRoute, (container) => {
```

Comment:
tree shaking related comment
```suggestion
export const RouteCallback = /*@__PURE__*/ Registration.cachedCallback(IRoute, (container) => {
```